### ISSUE TITLE
[Bug Report]: create more than some count multiple compiler rspack break

### ISSUE THREAD

**Issue by shaodahong** [web-infra-dev/rspack#2585]
### System Info

System:
    OS: Linux 5.15 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (2) x64 AMD EPYC
    Memory: 628.22 MB / 1.63 GB
    Container: Yes
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 16.17.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.15.0 - /usr/local/bin/npm
    Watchman: 20220528.183901.0 - /usr/local/bin/watchman
  npmPackages:
    @rspack/cli: ^0.1.1 => 0.1.1 
    @rspack/core: ^0.1.5 => 0.1.5 
    @rspack/dev-server: ^0.1.5 => 0.1.5

### Details

local hava many package, when i create more than some count multiple compiler, rspack break

```
[Error: error[internal]: Failed to resolve render_manifest results: JoinError::Panic(Id(123), ...).
Raw: panicked at 'Failed to resolve render_manifest results: JoinError::Panic(Id(123), ...)', /build/crates/rspack_core/src/compiler/compilation.rs:926:8
This is not expected, please file an issue at https://github.com/web-infra-dev/rspack/issues.

   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: start_thread
  27: clone

] {
  code: 'GenericFailure'
}
undefined
```

### Reproduce link

https://codesandbox.io/p/sandbox/snowy-cdn-cgr6t5?file=%2Fbuild.js&selection=%5B%7B%22endColumn%22%3A25%2C%22endLineNumber%22%3A26%2C%22startColumn%22%3A9%2C%22startLineNumber%22%3A26%7D%5D

### Reproduce Steps

1. `pnpm i`
2. `node build.js`

**Comment by JSerFeng** [comment]
You start the first compilation here (because you provide the second parameter which will lead to start running immediately:
<img width="605" alt="image" src="https://user-images.githubusercontent.com/57202839/229440141-90c4ca6d-f662-4ee0-a1da-d9a22ce00821.png">
And the server will start compilation again, so it is expected to fail

However now we do have problems when using multiple compiler instances, we are working on fixing this, you can temporarily use the following methods to make the compilation correct.
<img width="263" alt="image" src="https://user-images.githubusercontent.com/57202839/229441262-4d8744e6-e1a5-425f-94be-90f01a3ecb5e.png">

You can check it here: https://codesandbox.io/p/sandbox/friendly-pine-7yh7h0?file=%2Fpackage.json&selection=%5B%7B%22endColumn%22%3A25%2C%22endLineNumber%22%3A7%2C%22startColumn%22%3A25%2C%22startLineNumber%22%3A7%7D%5D

**Comment by JSerFeng** [comment]
> get a new error

I'll fix this.

>  use parallelism option unstable, sometimes still break,

works as expected for about 20 times in the sandbox, can you provide the error message or can you reproduce that in the demo ?

**Comment by shaodahong** [comment]
@JSerFeng still in my Reproduce link

<img width="763" alt="image" src="https://user-images.githubusercontent.com/14831261/229448274-70e2cccd-4cc1-4e0e-aeb9-5ca47b835ecf.png">

**Comment by JSerFeng** [comment]
Yes, this is expected to fail in your demo, because compiler can only be run once at a time, you start compilation here

<img width="413" alt="image" src="https://user-images.githubusercontent.com/57202839/229448439-05a58004-6f9b-406b-a2c1-acadf34a81ff.png">

And the server will start compilation again, so it failed, you should just make a compiler instead of running it at the beginning, like `rspack.rspack(configMaps)` instead of `rspack.rspack(configMaps, callback)`

**Comment by shaodahong** [comment]
@JSerFeng It doesn't have to appear, but it should appear if you try a few more times

**Comment by shaodahong** [comment]
> Yes, this is expected to fail in your demo, because compiler can only run once at a time, you start compilation here
> 
> <img alt="image" width="506" src="https://user-images.githubusercontent.com/57202839/229449302-0a8cc7e8-d8ae-468e-87aa-6802c9c8b250.png">
> 
> And the server will start compilation again, so it failed, you should just make a compiler instead of running it at the beginning, like `rspack.rspack(configMaps)` instead of `rspack.rspack(configMaps, callback)`

It ok, only this error

<img width="773" alt="image" src="https://user-images.githubusercontent.com/14831261/229449782-e263ca2a-1f17-4524-807c-0e78ccff7ca7.png">

**Comment by JSerFeng** [comment]
Thanks, I'll fix this when I got time

**Comment by shaodahong** [comment]
> Thanks, I'll fix this when I got time

@JSerFeng I find it's fixed in `0.1.7`, but when upgrade `0.1.7`, still error

```
➜  workspace git:(master) ✗ node build.js
<i> [webpack-dev-server] Project is running at:
<i> [webpack-dev-server] Loopback: http://localhost:8081/
<i> [webpack-dev-server] On Your Network (IPv4): http://192.168.241.2:8081/
<i> [webpack-dev-server] Content not from webpack is served from '/project/home/shaodahong/workspace/public' directory
[1]    5686 segmentation fault  node build.js
```

**Comment by JSerFeng** [comment]
Oh yes, I just fixed the `parallelism` part, but too many configs still crash the entire program, I'm still working on it

**Comment by JSerFeng** [comment]
Fixed, you can find the cause in #2744 , it was our carelessness
